### PR TITLE
Feature: Apprentice table Modal

### PR DESCRIPTION
**Pull request by OliwierOST**
Resolves: #185, #186, #184

Feature: Add apprentice table as modal, connect it to typesense add searching and filtering features as well as adding apprentices as favorites feature.

 
image

### CODE REVIEW

**Comment by OliwierOST** [comment]
This PR is on hold due to typesense licence expiry.

File: components/ApprenticeTablePanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const [showFilters, setShowFilters] = useState(false)
+  const { data: user } = useUser()
+
+  const transformItems = (items: Item[]) => {
+    console.log(items)
+    items.filter((element: Item) => element.value.includes(user?.uid))
+    if (items.length) {
+      items[0].label = "Favorites"
+    } else {
+      items.push({
```
[the remark this answers is not in the document]
Comment (OliwierOST) [reply components/ApprenticeTablePanel/index.tsx:29]: If you're asking if we're storing the favorites in the firestore then yes.

File: components/ApprenticeFilters/RangeInput.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      setValue({ min: stringMin, max: stringMax })
+    }
+  }, [currentRefinement])
+
+  useEffect(() => {
+    if (value.min == "") {
+      refine({ ...currentRefinement, min: min })
+    }
+    if (value.max == "") {
+      refine({ ...currentRefinement, max: max })
```
[the remark this answers is not in the document]
Comment (OliwierOST) [reply components/ApprenticeFilters/RangeInput.tsx:45]: Checked and everything seems to be in order.

**Review by piotrostr — APPROVED** [review]

**Review by OliwierOST — COMMENTED** [review]
LGTM

**Pull request merged**